### ISSUE TITLE
contraintGraph fails when running contraint_graph with self-loops

### ISSUE THREAD

**Issue by randy-marsh** [jmschrei/pomegranate#452]
I was messing around with constraint graphs  with the [constraint graph](https://github.com/jmschrei/pomegranate/blob/master/tutorials/Tutorial_4b_Bayesian_Network_Structure_Learning.ipynb ) 
tuto. but fails when i'm trying to add self loops:

`a.add_edge(c, b)
a.add_edge(c, d)
a.add_edge(b,b)`

when i run it with 'exact' algorithm throws:

> TypeError: Expected tuple, got int

best regards,

Randy
P.S. i'm running with windows 10, python 2.7 Ipython 5.4.1 and pomegranate 0.9.0

**Comment by jkleckner** [comment]
This seems a duplicate of #325 in which the workaround was to use one of the other algorithms different from `exact`.

**Comment by poplarShift** [comment]
Still not possible in 0.10.0 (which apparently fixed #325), but the error message has changed from what @randy-marsh reported. This one looks potentially more useful.

> TypeError: discrete_exact_a_star() takes exactly 6 positional arguments (7 given)

The workaround still works (i.e. with self-loops and e.g. algorithm='exact-dp').
(The constraint graph topology does not always seem to be respected though, but I will have to look more closely into that.)

Traceback:
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-12-4df2fade660c> in <module>
     29 m.plot()
     30 
---> 31 m = BayesianNetwork.from_samples(r, algorithm='exact',constraint_graph=c_self)
     32 m.plot()

pomegranate/BayesianNetwork.pyx in pomegranate.BayesianNetwork.BayesianNetwork.from_samples()

pomegranate/BayesianNetwork.pyx in pomegranate.BayesianNetwork.discrete_exact_with_constraints()

pomegranate/BayesianNetwork.pyx in pomegranate.BayesianNetwork.discrete_exact_with_constraints()

~/anaconda3/envs/trawl-dev/lib/python3.6/site-packages/joblib/parallel.py in __call__(self, iterable)
    984                 self._iterating = self._original_iterator is not None
    985 
--> 986             while self.dispatch_one_batch(iterator):
    987                 pass
    988 

~/anaconda3/envs/trawl-dev/lib/python3.6/site-packages/joblib/parallel.py in dispatch_one_batch(self, iterator)
    823                 return False
    824             else:
--> 825                 self._dispatch(tasks)
    826                 return True
    827 

~/anaconda3/envs/trawl-dev/lib/python3.6/site-packages/joblib/parallel.py in _dispatch(self, batch)
    780         with self._lock:
    781             job_idx = len(self._jobs)
--> 782             job = self._backend.apply_async(batch, callback=cb)
    783             # A job can complete so quickly than its callback is
    784             # called before we get here, causing self._jobs to

~/anaconda3/envs/trawl-dev/lib/python3.6/site-packages/joblib/_parallel_backends.py in apply_async(self, func, callback)
    180     def apply_async(self, func, callback=None):
    181         """Schedule a func to be run"""
--> 182         result = ImmediateResult(func)
    183         if callback:
    184             callback(result)

~/anaconda3/envs/trawl-dev/lib/python3.6/site-packages/joblib/_parallel_backends.py in __init__(self, batch)
    543         # Don't delay the application, to avoid keeping the input
    544         # arguments in memory
--> 545         self.results = batch()
    546 
    547     def get(self):

~/anaconda3/envs/trawl-dev/lib/python3.6/site-packages/joblib/parallel.py in __call__(self)
    259         with parallel_backend(self._backend):
    260             return [func(*args, **kwargs)
--> 261                     for func, args, kwargs in self.items]
    262 
    263     def __len__(self):

~/anaconda3/envs/trawl-dev/lib/python3.6/site-packages/joblib/parallel.py in <listcomp>(.0)
    259         with parallel_backend(self._backend):
    260             return [func(*args, **kwargs)
--> 261                     for func, args, kwargs in self.items]
    262 
    263     def __len__(self):

pomegranate/BayesianNetwork.pyx in pomegranate.BayesianNetwork.discrete_exact_with_constraints_task()

pomegranate/BayesianNetwork.pyx in pomegranate.BayesianNetwork.discrete_exact_a_star()

TypeError: discrete_exact_a_star() takes exactly 6 positional arguments (7 given)
```

Code:
```
from pomegranate import BayesianNetwork
import numpy as np
import pomegranate as pg
import networkx as nx

print(pg.__version__)
print(nx.__version__)

c = nx.DiGraph()
a1= 0,
a2= 1,
a3= 2,
c.add_edges_from([(a1,a2),(a1,a3),(a2,a3)])
pg.utils.plot_networkx(c)

c_self = nx.DiGraph()
n1 = (0,1) 
n2 = (2,) 
c_self.add_edge(n1, n2)
c_self.add_edge(n1,n1)
pg.utils.plot_networkx(c_self)

# random data
r=np.random.binomial(1,0.5,size=(1000,2))
# r=np.c_[r,r[:,0]+10]
r=np.c_[r,2*r[:,0]-r[:,1]]

m = BayesianNetwork.from_samples(r, algorithm='exact-dp',constraint_graph=c_self)
m.plot()

m = BayesianNetwork.from_samples(r, algorithm='exact',constraint_graph=c_self)
m.plot()

m = BayesianNetwork.from_samples(r, algorithm='exact',constraint_graph=c)
m.plot()
```
